builders/interoperability/mrl.md

```diff
@@ -0,0 +1,152 @@
+    ```
+
+To generate the second end of the transaction (remote EVM call), you need both the EVM transaction and the XCM instructions that executes said EVM transaction. The EVM transaction can be constructed as a transaction that interacts with the [batch precompile](/builders/pallets-precompiles/precompiles/batch){target=_blank} so that two transactions can happen in one. This is helpful because this EVM transaction has to both approve the xLabs relayer to relay the ERC-20 token as well as relay action itself.  
+
+??? code "Batched EVM Construction" 
+    ```js
+    --8<-- 'code/mrl/evm-tx.js'
+    ```
+
+You would want to send a message such that the `Transact` XCM instruction gets successfully executed. The most common method to do this is through `polkadotXcm.send` and sending the `WithdrawAsset`, `BuyExecution`, and `Transact` instructions. `RefundSurplus` and `DepositAsset` can also be used to ensure no assets get trapped, but they are technically optional.
+
+??? code "polkadotXcm.send Extrinsic Construction" 
+    ```js
+    --8<-- 'code/mrl/polkadotxcm-send.js'
+    ```
+
+To ensure that both the `xTokens.transferMultiassets` and the `polkadotXcm.send` transactions are sent together, you can batch them together using `utility.batchAll`. At time of writing, this helps ensure that the assets transfer happens before the EVM transaction, a necessary distinction. Unfortunately, this is subject to change with future XCM updates.  
+
+??? code "utility.batchAll Extrinsic Construction and Execution"
+    ```js
+    const batchExtrinsic = originChainPolkadotJsAPI.tx.utility.batchAll([
+        xTokensExtrinsic,
+        xcmExtrinsic
+    ]);
+
+    // Send batch transaction
+    return await batchExtrinsic.signAndSend(account, ({ status }) => {
+        if (status.isInBlock) console.log(`Moonbase Beta transaction finished!`);
+    });
+    ```
+
+If you would like to see an example script that fully implements this, an example is available in a [GitHub repository](https://github.com/jboetticher/mrl-reverse){target=_blank}.  
+
+It’s important to note that not every parachain will have xTokens and the other pallets implemented in a way that will allow this path. Substrate-based chains are very flexible, to the point where a standard doesn’t exist. If you believe your parachain does not support this path, please provide an alternative solution in the Moonbeam forum and to the Wormhole team.  
+
+### Tokens Available through Wormhole {: #tokens-available-through-wormhole }
+
+While Wormhole has the technical capability to bridge any token across chains, relayers will not support every token for fees. The ERC-20 assets that can be bridged through Wormhole's MRL solution are dependent on the tokens that the [xLabs relayer](https://xlabs.xyz/){target=_blank} takes in. The tokens that are available to Moonbeam are listed in the table below:  
+
+| Token Name |                  Address                   |
```

Comment:
these are just addresses for Moonbeam, right? what about Moonriver and Moonbase?

Reply:
I don't believe Moonriver is supported. Unsure of what is supported for Moonbase Alpha, as their relayer doesn't necessarily do testnet transactions in my experience. I have asked.